### PR TITLE
Add status upload file feature

### LINKED ISSUE #5832: as a end user I would like to be able to upload images to my status update on gitcoin, so i can share memes and other images
### User Story

[comment]: # (As a <user type>, I want to <task> so that <goal>.)
as a end user I would like to be able to upload images to my status update on gitcoin, so i can share memes and other images

### Why Is this Needed

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*:
people are visual

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*:
feature

### Current Behavior
[comment]: # (Describe what actually happened.)
no image upload

### Expected Behavior
[comment]: # (Describe what you expected to happen.)
user can upload an image via the status update field on gitcoin.co
image is uploaded to s3 and attached to the status update post
if that image is flagged by >3 people or an admin, then it is hidden + the user is not able to upload images anymore

### Definition of Done
[comment]: # (Any other information that would be useful, bullets are helpful.)
see above

### Additional Information
[comment]: # (Any other information that would be useful, content, screenshots, etc.)

48 hour bounty challenge - i want a PR this weekend!

### PR DESCRIPTION
**Pull request by iamonuwa**
##### Description
Allow users to upload images as part of status

##### Refers/Fixes
Fixes #5832 

##### Testing
Check the user profile, try uploading an image alongside the status or check the townsquare status box

### CODE REVIEW

**Comment by iamonuwa** [comment]
@octavioamu @owocki @danlipert, how do you want this to look like? **if that image is flagged by >3 people or an admin, then it is hidden + the user is not able to upload images anymore**

AFAIK, the flagging haapens on each of the updates

File: app/assets/v2/js/status.js [hunk 1]
```diff
@@ -55,12 +55,14 @@ $(document).ready(function() {
   function submitStatusUpdate() {
     const data = new FormData();
     const message = $('#textarea');
+    const file = $("#file-upload").prop("files")[0];
```
Comment (thelostone-mc) [inline app/assets/v2/js/status.js]: `file-upload` -> could we add a size check 
Don't want stuff crashing at our end (same size as grant image upload )

File: app/assets/v2/css/profile.css [hunk 2]
```diff
@@ -589,3 +589,24 @@ nav.navbar.navbar-dark{
...
+.profile-file-upload {
+  border: 1px solid #ccc;
+  display: inline-block;
+  padding: 6px 12px;
+  cursor: pointer;
+}
+
+input[type="file"] {
+  display: none;
+}
```
Comment (thelostone-mc) [inline app/assets/v2/css/profile.css]: ^ remove duplicate css

File: app/assets/v2/css/profile.css [hunk 3]
```diff
@@ -589,3 +589,24 @@ nav.navbar.navbar-dark{
...
+.custom-file-upload {
```
Comment (thelostone-mc) [inline app/assets/v2/css/profile.css]: this isn't used anywhere

File: app/assets/v2/js/status.js (hunk 1 above)
Comment (iamonuwa) [reply app/assets/v2/js/status.js]: I made it 3 MB. Not sure what the grant image upload size is.

**Comment by iamonuwa** [comment]
screen-capture-_9_

**Comment by iamonuwa** [comment]
> remove migrations and recreate as one neat migration + left a few other comments
> Could you add a recording of the PR

Removing the migrations and creating again yields same result

**Comment by androolloyd** [comment]
> > remove migrations and recreate as one neat migration + left a few other comments
> > Could you add a recording of the PR
> 
> Removing the migrations and creating again yields same result

The issue is that a merge had to be done which makes for a non linear path on migrations. 

If you remove the migrations, rebase onto master and recreate the migrations you should get a single file created. 

If not ping me on chat and we can figure it out, I have had to go through this a number of times already.

**Comment by danlipert** [comment]
@iamonuwa do the images get shown anywhere in the feed or will that be a separate PR? I'd like to see that functionality together

**Comment by iamonuwa** [comment]
Separate PR, yes. The spec of this PR was to allow status upload

File: app/dashboard/models.py [hunk 4]
```diff
@@ -1992,6 +1992,7 @@ class Activity(SuperModel):
     created = models.DateTimeField(auto_now_add=True, blank=True, null=True, db_index=True)
     activity_type = models.CharField(max_length=50, choices=ACTIVITY_TYPES, blank=True, db_index=True)
     metadata = JSONField(default=dict)
+    file= models.FileField(upload_to=get_upload_filename, null=True, blank=True, help_text=_('The image (jpeg, png, gif) url.'))
```
Comment (owocki) [inline app/dashboard/models.py]: will this upload to s3?

**Comment by owocki** [comment]
what does the image look like when its been posted to the feed? does it look good on mobile/desktop/tablet?

**Comment by iamonuwa** [comment]
@owocki, it uploads to s3

**Comment by iamonuwa** [comment]
@owocki, this pr uploads images with the status but doesn't display I.

**Comment by owocki** [comment]
> and attached to the status update post

ah ok when i said this i meant that it should be displayed :)

File: app/dashboard/models.py (hunk 4 above)
Comment (iamonuwa) [reply app/dashboard/models.py]: Yes, this does.

**Comment by iamonuwa** [comment]
> > and attached to the status update post
> 
> ah ok when i said this i meant that it should be displayed :)

Am going to show image if status has an image attached.

**Comment by owocki** [comment]
how goes? we get this feature requested regularly..  can we get it mergeable?

**Comment by iamonuwa** [comment]
> how goes? we get this feature requested regularly.. can we get it mergeable?

waiting for reviews

**Comment by danlipert** [comment]
@iamonuwa does this PR now show the uploaded image or is that coming in a followup PR?

**Comment by iamonuwa** [comment]
Follow up PR. This weekend. Everywhere is hot right now with the pandemic.

**Comment by danlipert** [comment]
> Follow up PR. This weekend. Everywhere is hot right now with the pandemic.

no worries man - stay safe

**Comment by thelostone-mc** [comment]
@iamonuwa looks like this PR has gotten stale! Could you throw in a fresh PR whenever you get around to this & we can get this in ?
Closing it out for now (reopen it -> if you'll be working on it / got a solution already ready based on the feedback comments)

**Pull request closed without merging**